lxd/db/cluster/entity_type_instance_snapshot.go

```diff
@@ -0,0 +1,59 @@
+package cluster
+
+import (
+	"fmt"
+)
+
+// entityTypeInstanceSnapshot implements entityTypeDBInfo for an InstanceSnapshot.
+type entityTypeInstanceSnapshot struct{}
+
+func (e entityTypeInstanceSnapshot) code() int64 {
+	return entityTypeCodeInstanceSnapshot
+}
+
+func (e entityTypeInstanceSnapshot) allURLsQuery() string {
+	return fmt.Sprintf(`
+SELECT %d, instances_snapshots.id, projects.name, '', json_array(instances.name, instances_snapshots.name) 
+FROM instances_snapshots 
+JOIN instances ON instances_snapshots.instance_id = instances.id 
+JOIN projects ON instances.project_id = projects.id`, e.code())
+}
+
+func (e entityTypeInstanceSnapshot) urlsByProjectQuery() string {
+	return fmt.Sprintf(`%s WHERE projects.name = ?`, e.allURLsQuery())
+}
+
+func (e entityTypeInstanceSnapshot) urlByIDQuery() string {
+	return fmt.Sprintf(`%s WHERE projects.name = ?`, e.allURLsQuery())
+}
+
+func (e entityTypeInstanceSnapshot) idFromURLQuery() string {
+	return `
+SELECT ?, instances_snapshots.id 
+FROM instances_snapshots 
+JOIN instances ON instances_snapshots.instance_id = instances.id 
+JOIN projects ON instances.project_id = projects.id 
+WHERE projects.name = ? 
+	AND '' = ? 
+	AND instances.name = ? 
+	AND instances_snapshots.name = ?`
+}
+
+func (e entityTypeInstanceSnapshot) onDeleteTriggerName() string {
+	return "on_instance_snaphot_delete" // TODO: Spelling was wrong originally. We need a patch to fix this.
```

Comment:
please can you open a GH issue for that to track it, if not already.

Reply:
https://github.com/canonical/lxd/issues/13862